### PR TITLE
fix(docs): Standardize jsdoc param/returns to use {Capital} for types.

### LINKED ISSUE #4225: Mixture of {Upper and {Lower case in jsdoc
Long standing gripe, @vladikoff pointed it out in one of my PRs 'cause I was confused about which to use when the file used both. Consistency brings inner peace.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
fixes #4225 

@philbooth - want to review 253 lines of boringness?

### CODE REVIEW

File: app/scripts/views/base.js [hunk 1]
```diff
@@ -739,9 +739,9 @@ define(function (require, exports, module) {
     /**
      * Place the cursor at the given position within the input element
      *
-     * @param {selector | element} which
-     * @param {number} selectionStart - defaults to after the last character.
-     * @param {number} selectionEnd - defaults to selectionStart.
+     * @param {Selector | element} which
```
Comment (philbooth) [inline app/scripts/views/base.js]: Should we capitalise to `Element` here too?

File: app/scripts/models/account.js [hunk 2]
```diff
@@ -605,9 +605,9 @@ define(function (require, exports, module) {
      * Check whether all the passed in permissions have been
      * seen previously.
      *
-     * @param {string} clientId
-     * @param {strings[]} permissions
-     * @returns {boolean} `true` if client has seen all the permissions,
+     * @param {String} clientId
+     * @param {Strings[]} permissions
```
Comment (philbooth) [inline app/scripts/models/account.js]: I realise you're just capitalising what already existed here, but this notation feels a bit funny to me. Just wondering whether something like `@param {Array[String]}` would be less prone to mis-reading at a glance. What do you reckon?

File: app/scripts/views/base.js (hunk 1 above)
Comment (philbooth) [inline app/scripts/views/base.js]: Also, is `Selector` the correct type here or should it really be `String`? `Selector` feels more like it belongs in the description to me.

(and sorry, I know you haven't changed this bit)

**Comment by shane-tomlinson** [comment]
Here's the script I used to make the changes:


```bash
#!/bin/bash

PREFIXES[1]="\@param \{" 
PREFIXES[2]="\@returns \{"
PREFIXES[3]="\@return \{"

for i in 1 2 3
do
  PREFIX="${PREFIXES[i]}"
  for LOWER in {a..z}
    do
      UPPER=`echo "${LOWER}" | tr '[:lower:]' '[:upper:]'`
      echo ${PREFIX}${LOWER} ${PREFIX}${UPPER}

      sr "${PREFIX}${LOWER}" "${PREFIX}${UPPER}" *.js
    done
done

sr "{Undefined" "{undefined" *.js
```

File: app/scripts/views/base.js [hunk 3]
```diff
@@ -838,7 +838,7 @@ define(function (require, exports, module) {
      *
      * @method invokeBehavior
      * @param {Function} behavior
-     * @returns {variant} behavior's return value if behavior is a function,
+     * @returns {Variant} behavior's return value if behavior is a function,
```
Comment (philbooth) [inline app/scripts/views/base.js]: Again, I realise you're only capitalising what's already here but `Variant` reminds me of the COM type. Maybe that's fine but I find it a little bit jarring.

**Comment by philbooth** [comment]
@shane-tomlinson there's some comments inline but you're welcome to ignore them. r+

File: app/scripts/models/account.js (hunk 2 above)
Comment (shane-tomlinson) [inline app/scripts/models/account.js]: I prefer your suggestion, though it goes against the [jsdoc spec](http://usejsdoc.org/tags-type.html):

```

An array of MyClass instances.

{Array.<MyClass>}
// or:
{MyClass[]}
```

Comment (philbooth) [inline app/scripts/models/account.js]: Oh, fair enough. Does it mean our trailing 's' on `Strings` is wrong in that case?

Comment (philbooth) [inline app/scripts/models/account.js]: (although that is even more mis-readable so whatevs)

**Comment by shane-tomlinson** [comment]
@philbooth - I realized I missed lots of directories, my `find` foo is not as good as I had hoped.

**Comment by philbooth** [comment]
> I realized I missed lots of directories

Uh oh...

**Pull request merged**